### DISCUSSION TITLE
[IntelliSense] Haml support

### DISCUSSION THREAD

**Discussion by alDuncanson** [tailwindlabs/tailwindcss#7584] [Ideas]
Are there plans for supporting `*.haml` files?

I tried to use the **Include Languages** setting:
<img width="766" alt="image" src="https://user-images.githubusercontent.com/25286675/155189878-4d2da9a5-2b35-40b0-90cb-406dc9814672.png">

Am I doing something wrong or will the extension not work with `haml`?

**Comment by jaredmoody** [comment]
I've had some limited success mapping haml to html and then setting `tailwindCSS.experimental.classRegex` setting to pick up the class names.

**Comment by HaavGa** [comment]
[This](https://github.com/paolotiu/tailwind-intellisense-regex-list#haml) worked perfectly!

**Reply by DrTobe** [reply]
If I understand it correctly, the "mapping haml to html" step is required additionally to setting the `tailwindCSS.experimental.classRegex`. In my setup, doing this in User Settings did not work but it can be done in the Workspace Settings, too. In total, the following `.vscode/settings.json` works for me:
```json
{
  "tailwindCSS.includeLanguages": {
    "haml": "html"
  },
  "tailwindCSS.experimental.classRegex": [
      [ "class: ?\"([^\"]*)\"", "([a-zA-Z0-9\\-:]+)" ],
      [ "(\\.[\\w\\-.]+)[\\n\\=\\{\\s]", "([\\w\\-]+)" ],
    ]
}
```

Additionally, if you want to do this in neovim, the following lua config snippet works for me:
```lua
require 'lspconfig'.tailwindcss.setup({
    settings = {
        tailwindCSS = {
            includeLanguages = {
                haml = "html"
            },
            experimental = {
                classRegex = {
                    { "class: ?\"([^\"]*)\"", "([a-zA-Z0-9\\-:]+)" },
                    { "(\\.[\\w\\-.]+)[\\n\\=\\{\\s]", "([\\w\\-]+)" },
                }
            }
        }
    }
})
```

**Reply by Bakterio** [reply]
Thanks a lot, especially for the neovim settings!!!!
